unittests/linuxliketests.py

```diff
@@ -576,8 +576,6 @@ def test_installed_modes(self):
         Test that files installed by these tests have the correct permissions.
         Can't be an ordinary test because our installed_files.txt is very basic.
         '''
-        if is_cygwin():
-            self.new_builddir_in_tempdir()
```

Comment:
This commit message doesn't justify why it's okay to allow the umask tests to now run in the source tree by default.

Reply:
Because it doesn't run on the source tree, and that has nothing to do with the source tree? The source tree is always, unconditionally copied into a new directory, the only thing that changes here is where the build directory is being placed.

Reply:
This commit message doesn't justify why it's okay to allow the umask tests to now run in the meson.git source tree by default.

Reply:
And the reason I'm mentioning this is because the comment in the now deleted function says cygwin requires forcing a tempdir for this small subset of tests, it doesn't say anything about the relationship between the directory with meson.build and the one with build.ninja

Reply:
Okay, your comment confused me, I thought you were suggesting that the test is now configuring from the source tree, not that there were issues with the build directory being in the source tree.

I'm curious as to what the actual original issue with installing in the source dir was, and why it doesn't seem to be a problem now, looking at the code there was a CI regression on cygwin (when we were running cygwin on azure) from this, but it doesn't seem to happen now. I'm really confused

Reply:
I don't know either! :D 

If we think it's no longer relevant then I'm not opposed to deleting it, it's just the commit message should indicate we are doing so (and why).

Reply:
Okay, I've added a paragraph about those changes to the commit message.

Reply:
Interesting and interesting. This failure seems to be sporadic. I'm going to guess there's a race condition involved, but it sounds hard to solve, so I just more-or-less put the workaround back.